Update documentation for adding a New Grib Product

srcarter3 opened the pull request:
- update the existing new grid page to only include content about grib2
- added an entirely new, hidden, page for dealing with original grib product
- updated all "notes" and "warnings" with new admonition boxes
- restructured some of the sections and added helpful subsection titles
- added small new sections "Creating New Menu Item" and "Using wgrib2"
- removed the environment version dependencies for mkdocs and mkdocs-unidata (because I'm updating the mkdocs-unidata package as well right now)
- modified the mkdocs.yml file slightly to use new "nav" tag instead of "pages"

Commit:
Update documentation for adding a New Grib Product
- update the existing new grid page to only include content about grib2
- added an entirely new, hidden, page for dealing with original grib product
- updated all "notes" and "warnings" with new admonition boxes
- restructured some of the sections and added helpful subsection titles
- added small new sections "Creating New Menu Item" and "Using wgrib2"
- removed the environment version dependencies for mkdocs and mkdocs-unidata (because I'm updating the mkdocs-unidata package as well right now)
- modified the mkdocs.yml file slightly to use new "nav" tag instead of "pages"

Commit:
Merge branch 'unidata_18.2.1' of github.com:Unidata/awips2 into new-grib-doc-changes
Bring over GHA change that no longer restricts mkdocs version

Commit:
Add mkdocs and mkdocs-unidata as dependency but remove version restriction

Commit:
Merge pull request #2 from srcarter3/new-grib-doc-changes

New grib doc changes

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -15,105 +17,52 @@ If the parameters in the grib file haven't been previously specified, another ch
+    cp cpti.grib2 /awips2/data_store/ingest/
 
-3. Check that the hdf5 data directory exists for our unnamed grid
+Remember that the data distribution file (`/awips2/edex/data/utility/common_static/base/distribution/grib.xml`) will match filenames which have the `*.grib*` extension.
```
could add a link here:

[data distribution](/awips2/edex/data-distribution-files/)

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -15,105 +17,52 @@ If the parameters in the grib file haven't been previously specified, another ch
+### Download Test Data
 
-1. Download an example grib1 file and rename to a `*.grib` extension, then copy to the manual ingest point `/awips2/data_store/ingest/` 
+Download an example grib2 file (make sure the extension is `.grib2` or EDEX may not recognize it), and then copy to the manual ingest point `/awips2/data_store/ingest/` 
```
"(make sure the extension is `.grib2` or EDEX may not recognize it)"
and 
"Remember that the data distribution file..."  (line 28)

is pretty repetitive, I suggest combining those two sentences

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -15,105 +17,52 @@ If the parameters in the grib file haven't been previously specified, another ch
-### Grib2 Products
-
-1. Download an example grib2 file and rename to a `*.grib2` extension, then copy to the manual ingest point `/awips2/data_store/ingest/` 
+    edex log grib
```
I know this was in here originally, but I think the user would have to run that before they ingest the file. It might be better to tell them to just look at the log file (/awips2/edex/logs/edex-ingestGrib-YYYYMMDD.log)

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -15,105 +17,52 @@ If the parameters in the grib file haven't been previously specified, another ch
         ls -latr /awips2/edex/data/hdf5/grid/GribModel:161:0:97
 
-    Though the grib file has been decoded, it has been given a generic name with its center, subcenter, and process IDs (161, 0, 97, respectively). 
+    Though the grib file has been decoded, it has been given a generic name with its **center, subcenter, and process IDs** (161, 0, 97, respectively). 
```
check the formatting on this, it's currently in the "code block" 
![image](https://user-images.githubusercontent.com/65614396/162507657-a92b33cf-4197-4cd2-9360-b1b503b61ae0.png)

Commit:
Small doc changes for New Grid page, based on Tiffany's review

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -144,12 +93,16 @@ Notice that our grib2 file has a **Lat/lon Grid** projection.  Where:
 ## Create Grid Projection File
     
+### Projection Types
+    
```
For the projection types, maybe make a "Note":

You may not have information for every tag listed, for example it's not required for the latLonGridCoverage to have spacingUnit, la2, lo2

This was something Pete made a note about when referencing our documentation

srcarter3 replied:
![image](https://user-images.githubusercontent.com/21114460/162518232-6efb46fc-d2a6-451a-b38b-2b542d37ab14.png)


Is this better?

tiffanycmeyer13 replied:
Yep, that looks good!

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -283,48 +203,14 @@ And edit the new `cpti.xml` to define the projection values using the [output fr
     </latLonGridCoverage>
   
-> **Note**: Notice the `<name>384000</name>` tag was defined from the number of grid points (600 and 640). This value will be matched against an entry in our models file (below) to set the name of the model (e.g. CPTI).
+!!! note "Notice the `<name>384000</name>` tag was created by using the number of grid points (600 and 640). This name can be anything as long as it is unique and will be used to match against in the model definition."
```
To match what's in the output use `<name>600640</name>`

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -228,45 +179,14 @@ Grid projection files are stored in `/awips2/edex/data/utility/common_static/bas
 Copy an existing xml file with the same grid projection type (in this case **latLonGridCoverage**) to a new file `cpti.xml`:
 
     cd /awips2/edex/data/utility/common_static/base/grib/grids/
     cp MRMS-1km.xml cpti.xml
```
The MRMS file name has changed, it's now MRMS-1km-CONUS.xml

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -464,10 +316,24 @@ You will have to restart ingestGrib for the changes to take place:
+
+## Using wgrib2
+
+Mentioned in this page are a few command parameters for `wgrib2` such as `-center`, `-subcenter`, and `-full_name`.
```
add -grid since that was used as well.

srcarter3 replied:
also added `-varX` because we reference that one too.

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -15,105 +17,50 @@ If the parameters in the grib file haven't been previously specified, another ch
+    ...
     
-    Remember that the data distribution file (`/awips2/edex/data/utility/common_static/base/distribution/grib.xml`) will match filenames which have the `*.grib*` extension.
+!!! warning "This step may fail if the parameter is not yet defined.  The error will look like:"
```
Specifically for the cpti.grib2 file - the parameters don't exist so you do need to create the tables first before you can actually see it ingested. I know you have the note in here, but maybe explicitly state in this specific case, we have to define the parameters first.

srcarter3 replied:
I think the idea (similar to what Jason Burke's does at times) is to have the user purposely encounter the error, so they're familiar with it.

How is this:
![image](https://user-images.githubusercontent.com/21114460/162518662-70d6c844-486a-4099-928e-b2a89679639a.png)

tiffanycmeyer13 replied:
Agreed. 

The only other thing to maybe mention, is in the second code block, it shows that it was "ingested" (line 2) but it really wasn't.

tiffanycmeyer13 commented:
We should also add a snippet here (you had a small section in the blog) about the default parameters for most models being in tables/-1/-1 instead of tables/<center>/<subcenter>

tiffanycmeyer13 commented on docs/edex/new-grid.md:
```diff
@@ -15,105 +17,50 @@ If the parameters in the grib file haven't been previously specified, another ch
-### Grib2 Products
-
-When the grid was ingested a record was added to the `grid_coverage` table with its navigation information:
+When a grid is ingested a record is added to the `grid_coverage` table with its navigation information:
```
In the psql command, let's add a few more parameters. These are more common for the other projections, but can be helpful. It also prints out the firstGridPointCorner which can be helpful:

here's the new output:

```
metadata=> select nx,ny,dx,dy,majoraxis,minoraxis,la1,lo1,lov,latin1,latin2,spacingunit,lad,la2,latin,lo2,firstgridpointcorner from gridcoverage where id=(select distinct(location_id) from grid_info where datasetid='GribModel:161:0:97');
 nx  | ny  |  dx   |  dy   | majoraxis | minoraxis |    la1    | lo1 | lov | latin1 | latin2 | spacingunit | lad | la2 | latin | lo2 | firstgridpointcorner 
-----+-----+-------+-------+-----------+-----------+-----------+-----+-----+--------+--------+-------------+-----+-----+-------+-----+----------------------
 600 | 640 | 0.005 | 0.005 |           |           | 40.799999 | 261 |     |        |        | degree      |     |     |       |     | UpperLeft
(1 row)
```

srcarter3 commented:
> We should also add a snippet here (you had a small section in the blog) about the default parameters for most models being in tables/-1/-1 instead of tables/
> 
> /

How's this:
![image](https://user-images.githubusercontent.com/21114460/162521427-8d5a7f1b-bb05-4e6a-a86b-31ea21f0ab23.png)

Commit:
Small changes to New Grid doc based on Tiffany's review:
- update the table parameter warning because our example WILL encounter this error
- update the psql query and output with additional columns
- add a note for projection types that not all fields are necessary
- update the filename for our test data since it changed
- update the name tag in the note to match the name tag example above
- add a note about default parameter table location (.../-1/-1/)
- add a few more commands to the wgrib2 section to represent all commands used on this page
